dev/stream-state-gen-protos.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+set -ex
+
+if [[ $# -gt 0 ]]; then
```

Comment:
I have written a script with reference to  https://github.com/apache/spark/blob/master/dev/connect-gen-protos.sh , whose main function is to use buf to generate corresponding `_pb2.py` and `_pb2.pyi` files from `.proto` files in the `sql/core/src/main/protobuf/` directory. 

Currently, the OUTPUT_PATH is fixed to `${SPARK_HOME}/python/pyspark/sql/streaming/proto`. If there are more complex requirements in the future, this script will need to be further enhanced.

Reply:
With this pr, after modifying the proto file, we can execute `dev/stream-state-gen-protos.sh` to regenerate the required `_pb2.py` and `_pb2.pyi` files and then submit them.